flambe/compile/serialization.py

```diff
@@ -97,12 +101,27 @@ def _update_save_tree(save_tree: SaveTreeNode, key: Sequence[str], value: Any) -
     last_i = 0
     for i, _ in enumerate(key):
         current_key = STATE_DICT_DELIMETER.join(key[last_i:i + 1])
+        # current_state_key = key[i]
         if current_key in current.children:
             current = current.children[current_key]  # type: ignore
-            last_i += 1
+            last_i = i+1
+            current_key = STATE_DICT_DELIMETER.join(key[last_i:i + 1])
     current.state[current_key] = value
 
 
+def _update_save_tree_metadata(save_tree: SaveTreeNode, key: Sequence[str], value: Any) -> None:
+    current = save_tree
+    last_i = 0
+    for i, _ in enumerate(key):
+        current_key = STATE_DICT_DELIMETER.join(key[last_i:i + 1])
+        # current_state_key = key[i]
+        if current_key in current.children:
+            current = current.children[current_key]  # type: ignore
+            last_i = i+1
+            current_key = STATE_DICT_DELIMETER.join(key[last_i:i + 1])
```

Comment:
This has a lot of duplicated code with `_update_save_tree`. Can we extract this logic in a single method?